src/components/modebar/buttons.js

```diff
@@ -348,10 +348,14 @@ function handleCamera3d(gd, ev) {
         var scene = fullLayout[sceneId]._scene;
 
         if(attr === 'resetDefault' || attr === 'resetLastSave') {
-            aobj[key] = Lib.extendDeep({}, scene.viewInitial);
-
-            aobj[key].projection = {
-                type: (scene.camera._ortho) ? 'orthographic' : 'perspective'
+            aobj[key] = {
+                'up': scene.viewInitial.up,
+                'eye': scene.viewInitial.eye,
+                'center': scene.viewInitial.center,
+                'projection': {
```

Comment:
We shouldn't need to update `projection` at all, am I right?

Reply:
You are right. But in order to that we need to explicitly pass the current projection here.

Reply:
Ok, that's a bug then. Calling

```
Plotly.relayout('graph', 'scene.camera.up', {/**/})
```

should not change the projection type.

Reply:
don't we just need to flatten it to `aobj[key + '.up'] = ...` etc?

Reply:
Thanks @alexcjohnson & @etpinard.
Flattening was really helpful.